File: kala-game-service/src/main/java/org/jesperancinha/games/kalagameservice/model/Player.java
package org.jesperancinha.games.kalagameservice.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.OneToOne;
import javax.persistence.Table;
import java.util.List;
import java.util.Objects;

@Entity
@Table
@Setter
@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Player {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(unique = true)
    private String username;

    @OneToMany(fetch = FetchType.EAGER)
    private List<Board> boards;

    @OneToOne
    private Player opponent;

    @OneToOne
    private Board currentBoard;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Player)) return false;
        Player player = (Player) o;
        return Objects.equals(getUsername(), player.getUsername());
    }

    @Override
    public int hashCode() {
        return Objects.hash(getUsername());
    }
}

File: kala-game-service/src/main/kotlin/org/jesperancinha/games/kalagameservice/model/Player.kt
package org.jesperancinha.games.kalagameservice.model

import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.FetchType
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.OneToMany
import javax.persistence.OneToOne
import javax.persistence.Table

@Entity
@Table
data class Player(
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    internal val id: Long? = null,

    @Column(unique = true)
    internal val username: String? = null,

    @OneToMany(fetch = FetchType.EAGER)
    internal var boards: MutableList<Board>? = null,

    @OneToOne
    internal var opponent: Player? = null,

    @OneToOne
    internal var currentBoard: Board? = null
)